### ISSUE TITLE
subset not working with negative coordinates

### ISSUE THREAD

**Issue by bwlewis** [Paradigm4/SciDBR#82]
Here is an example:

```r
iquery("create_array(MODIS_DATA,  <altitude:double> [latitude_e4=-900000:900000,20000,0,longitude_e4=-1800000:1800000,20000,0], 0)")> arr = scidb("MODIS_DATA")
lat1 = 1
lat2 = 2
long1 = -1
long2 = 2
subset(arr, latitude_e4 >= lat1 & latitude_e4 <= lat2 & longitude_e4 >= long1 & longitude_e4 <= long2)@name
```

The output includes a SciDB filter; it should only have SciDB between!

**Comment by bwlewis** [comment]
Only a partial fix. This expression stil doesn't work right:

```r
subset(arr, latitude_e4 >= lat1 & latitude_e4 <= lat2 & longitude_e4 >= -30 & longitude_e4 <= long2)@name

[1] "filter(between(MODIS_DATA,1,null,2,2),true  and longitude_e4 >= - 30 )"
```

**Comment by bwlewis** [comment]
Fixed now, commit 91fbd5b7b827eb4a0a2addb849eb7ea316d09638
